src/memory.rs

```diff
@@ -120,7 +120,7 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
             bytes: Vec::new(),
             relocations: BTreeMap::new(),
             undef_mask: UndefMask::new(0),
-            align: 1,
+            align: 8, // should be infinity?
```

Comment:
The rest of the PR looks good, but I'm also iffy on this change. If this alignment affects anything, we should just make it not affect those things.